File: calculator/calculator.go
package calculator
import (
	"fmt"
	"strings"
)


func Calculator() float64{
	var result float64 = 0
	var Result = &result // Used this just to get a feel of how pointers work.
	var num1 float64
	var num2 float64
	var sign string


	fmt.Print("Enter first number: ")
	fmt.Scanln(&num1)
	fmt.Print("Enter Sign, e.g X for multiplication: ")
	fmt.Scanln(&sign)
	fmt.Print("Enter second number: ")
	fmt.Scanln(&num2)

	*Result = subCal(num1, num2, sign)
	fmt.Println(result) // Prints out the initial result 

	// An infinite loop to continually run the calculator
	for {
		var fNum float64
		var sign string
		fmt.Print("Enter Sign: ")
		fmt.Scanln(&sign)
		var newSign = strings.ToLower(sign) // convert sign input to lower to check sign
		if newSign == "exit" {
			break
		}else{
		fmt.Print("Enter Number: ")
		fmt.Scanln(&fNum)
		*Result = subCal(*Result, fNum, sign)
		fmt.Println(result)
		}
	}
	return *Result
}


func subCal(num float64, fNum float64, sign string) float64 {

	var result float64

	switch sign {
	case "+":
		result = num + fNum
	case "-":
		result = num - fNum
	case "*", "X", "x":
		result = num * fNum
	case "/":
		result = num / fNum
	}
	return result
}

File: python/calculator.py
# Python equivalent to the Go calculator in the calculator package
# From Python 3.10 you can now use the switch statement
def subCal(num1, num2, sign):
    result = 0
    match sign:
        case "+":
            result = num1 + num2
        case "-":
            result = num1 - num2
        case "X" | "x" | "*":
            result = num1 * num2
        case "/":
            result = num1 / num2

    return result


def calculator():
    num1 = int(input("Enter first number: "))
    sign = input("Enter sign, e.g +, -, /: ")
    num2 = int(input("Enter second number: "))
    
    result = subCal(num1, num2, sign)
    print(result)
    
    while True:
        sign = input("Enter sign, e.g +, -, /: ")
        if sign.lower() == "exit":
            break
        else:
            num3 = int(input("Enter second number: "))
            result = subCal(result, num3, sign)
            print(result)
        
        
    
    